src/transformers/models/efficientformer/modeling_efficientformer.py

```diff
@@ -0,0 +1,641 @@
+        hidden_states = self.fc2(hidden_states)
+        hidden_states = self.drop2(hidden_states)
+
+        return hidden_states
+
+class EfficientformerConvMlp(nn.Module):
+
+    def __init__(self, config, in_features, hidden_features=None, out_features=None, drop=0.):
+        super().__init__()
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+
+        self.fc1 = nn.Conv2d(in_features, hidden_features, 1)
+        self.act = ACT2FN[config.hidden_act]
+        self.fc2 = nn.Conv2d(hidden_features, out_features, 1)
+        self.drop = nn.Dropout(drop)
+
+        self.norm1 = nn.BatchNorm2d(hidden_features)
+        self.norm2 = nn.BatchNorm2d(out_features)
+
+    def forward(self, x):
+        x = self.fc1(x)
+
+        x = self.norm1(x)
+
+        x = self.act(x)
+        x = self.drop(x)
+        x = self.fc2(x)
+
+        x = self.norm2(x)
+
+        x = self.drop(x)
+        return x
+
+# Copied from transformers.models.beit.modeling_beit.drop_path
+def drop_path(input, drop_prob=0.0, training=False, scale_by_keep=True):
+    """
+    Drop paths (Stochastic Depth) per sample (when applied in main path of residual blocks).
+    Comment by Ross Wightman: This is the same as the DropConnect impl I created for EfficientNet, etc networks,
+    however, the original name is misleading as 'Drop Connect' is a different form of dropout in a separate paper...
+    See discussion: https://github.com/tensorflow/tpu/issues/494#issuecomment-532968956 ... I've opted for changing the
+    layer and argument names to 'drop path' rather than mix DropConnect as a layer name and use 'survival rate' as the
+    argument.
+    """
+    if drop_prob == 0.0 or not training:
+        return input
+    keep_prob = 1 - drop_prob
+    shape = (input.shape[0],) + (1,) * (input.ndim - 1)  # work with diff dim tensors, not just 2D ConvNets
+    random_tensor = keep_prob + torch.rand(shape, dtype=input.dtype, device=input.device)
+    random_tensor.floor_()  # binarize
+    output = input.div(keep_prob) * random_tensor
+    return output
+
+
+# Copied from transformers.models.beit.modeling_beit.BeitDropPath with Beit->Efficientformer
+class EfficientformerDropPath(nn.Module):
+    """Drop paths (Stochastic Depth) per sample (when applied in main path of residual blocks)."""
+
+    def __init__(self, drop_prob: Optional[float] = None) -> None:
+        super().__init__()
+        self.drop_prob = drop_prob
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return drop_path(x, self.drop_prob, self.training)
+
+    def extra_repr(self) -> str:
+        return "p={}".format(self.drop_prob)
+
+class EfficientformerMeta3D(nn.Module):
+    
+    def __init__(self, config, dim, norm_layer=nn.LayerNorm, drop_path=0.):
+
+        super().__init__()
+
+        self.norm1 = norm_layer(dim, eps=config.layer_norm_eps)
+        self.token_mixer = EfficientformerSelfAttention(dim)
```

Comment:
Please see my comment for `EfficientformerSelfAttention`, we aim to not hardcode configuration values in order to support multiple versions of the same model.

```suggestion
        self.token_mixer = EfficientformerSelfAttention(
            dim=config.dim,
            key_dim=config.key_dim,
            num_heads=config.num_heads,
            ...
        )
```

These values should also be added to the configuration files so that the model can be correctly initialized.